### PR TITLE
[ja] Sync the Japanese Job page with the latest version

### PR DESCRIPTION
**Pull request by morix1500**
こちらのIssueを対応しました。https://github.com/kubernetes/website/issues/41364
差分多いですがレビューおねがいします 🙇

### CODE REVIEW

File: content/ja/docs/concepts/workloads/controllers/job.md [hunk 1]
```diff
@@ -272,13 +292,63 @@ Podがノードからキックされた(ノードがアップグレード、再
...
+Pod失敗ポリシーまたはPod失敗のバックオフポリシーのいずれかによってJobが失敗し、そのJobが複数のPodを実行している場合、KubernetesはそのJob内の保留中または実行中のすべてのPodを終了します。
+{{< /note >}}
+
+これらはAPIの要件と機能です:
+  - `.spec.podFailurePolicy`フィールドをJobに使いたい場合は、`.spec.restartPolicy`を`Never`に設定してそのJobのPodテンプレートも定義する必要があります。
+  - `spec.podFailurePolicy.rules`で指定したPod失敗ポリシーのルールが順番に評価されます。あるPodの失敗がルールに一致すると、残りのルールは無視されます。Pod失敗に一致するルールがない場合は、デフォルトの処理が適用されます。
+  - `spec.podFailurePolicy.rules[*].containerName`を指定することで、ルールを特定のコンテナに制限することができます。指定しない場合、ルールはすべてのコンテナに適用されます。指定する場合は、Pod テンプレート内のコンテナ名または`initContainer`名のいずれかに一致する必要があります。
+  - Pod失敗ポリシーが`spec.podFailurePolicy.rules[*].action`にマッチしたときに実行されるアクションを指定できます。指定可能な値は以下のとおりです。
+    - `FailJob`: PodのJobを`Failed`としてマークし、実行中の Pod をすべて終了させる必要があることを示します。 
+    - `Ignore`: `.spec.backoffLimit`のカウンタは加算されず、代替のPodが作成すべきであることを示します。
```
Comment (nasa9084) [inline content/ja/docs/concepts/workloads/controllers/job.md:338]:
```suggestion
    - `Ignore`: `.spec.backoffLimit`のカウンターは加算されず、代替のPodが作成すべきであることを示します。
```

File: content/ja/docs/concepts/workloads/controllers/job.md [hunk 2]
```diff
@@ -272,13 +292,63 @@ Podがノードからキックされた(ノードがアップグレード、再
...
+    - `Count`: Podがデフォルトの方法で処理されるべきであることを示します。`.spec.backoffLimit`のカウンタが加算されます。
```
Comment (nasa9084) [inline content/ja/docs/concepts/workloads/controllers/job.md:339]:
```suggestion
    - `Count`: Podがデフォルトの方法で処理されるべきであることを示します。`.spec.backoffLimit`のカウンターが加算されます。
```

**Review by nasa9084 — COMMENTED** [review]
/assign

File: content/ja/docs/concepts/workloads/controllers/job.md (hunk 1 above)
Comment (morix1500) [reply content/ja/docs/concepts/workloads/controllers/job.md]: @nasa9084 ありがとうございます！修正いたしました。

File: content/ja/docs/concepts/workloads/controllers/job.md (hunk 2 above)
Comment (morix1500) [reply content/ja/docs/concepts/workloads/controllers/job.md]: @nasa9084 ありがとうございます！修正いたしました。

**Review by nasa9084 — COMMENTED** [review]
/approve

File: content/ja/docs/concepts/workloads/controllers/job.md [hunk 3]
```diff
@@ -479,20 +566,15 @@ Events:
...
-デフォルトで有効になっています。
-{{< /note >}}
+{{< feature-state for_k8s_version="v1.27" state="stable" >}}
 
 ほとんどの場合、並列Jobは、すべてのPodが同じゾーン、またはすべてのGPUモデルxかyのいずれかであるが、両方の混在ではない、などの制約付きで実行することが望ましいです。
 
 [suspend](#suspending-a-job)フィールドは、これらの機能を実現するための第一歩です。Suspendは、カスタムキューコントローラーがJobをいつ開始すべきかを決定することができます。しかし、Jobの一時停止が解除されると、カスタムキューコントローラーは、Job内のPodの実際の配置場所には影響を与えません。
 
-この機能により、Jobが開始される前にスケジューリング命令を更新でき、カスタムキューコントローラーがPodの配置に影響を与えることができると同時に、実際のPodとノードの割り当てをkube-schedulerにオフロードすることができます。これは一時停止されたJobの中で、一度も一時停止解除されたことのないJobに対してのみ許可されます。
+この機能により、Jobが開始される前にスケジューリング命令を更新でき、カスタムキューコントローラーがPodの配置に影響を与えることができるようになります。これは一時停止されたJobの中で、一度も一時停止解除されたことのないJobに対してのみ許可されます。
```
Comment (atoato88) [inline content/ja/docs/concepts/workloads/controllers/job.md:575]: In `en` content, the sentence following is included:
- "while at the same time offloading actual pod-to-node assignment to kube-scheduler."
Could you add the translation about this?

https://github.com/kubernetes/website/blob/main/content/en/docs/concepts/workloads/controllers/job.md?plain=1#L751-L752

File: content/ja/docs/concepts/workloads/controllers/job.md [hunk 4]
```diff
@@ -538,34 +620,33 @@ spec:
...
+### 静的なインデックス付きJob {#elastic-indexed-jobs}
+
+{{< feature-state for_k8s_version="v1.27" state="beta" >}}
 
-クラスターで`JobTrackingWithFinalizers`機能を有効にすると、コントロールプレーンは任意のJobに属するPodを追跡し、そのようなPodがAPIサーバーから削除された場合に通知します。そのために、Jobコントローラーは`batch.kubernetes.io/job-tracking`Finalizerを持つPodを作成します。コントローラーはPodがJobステータスに計上された後にのみFinalizerを削除し、他のコントローラーやユーザーによるPodの削除を可能にします。
+`.spec.parallelism`と`.spec.compleitions`の両方を、`.spec.parallelism` == `.spec.compleitions`となるように変更することで、インデックス付きJobを増減させることができます。[APIサーバ](/docs/reference/command-line-tools-reference/kube-apiserver/)の`ElasticIndexedJob`[フィーチャーゲート](/ja/docs/reference/command-line-tools-reference/feature-gates/)が無効になっている場合、`.spec.compleitions`は不変です。
 
-Jobコントローラーは、新しいJobに対してのみ新しいアルゴリズムを使用します。この機能が有効になる前に作成されたJobは影響を受けません。JobコントローラーがPod FinalizerでJob追跡しているかどうかは、Jobが`batch.kubernetes.io/job-tracking`というアノテーションを持っているかどうかで判断できます。
-このアノテーションを手動で追加または削除しては**いけません**。
+静的ななインデックス付きJobの使用例としては、MPI、Horovord、Ray、PyTorchトレーニングジョブなど、インデックス付きJobのスケーリングを必要とするバッチワークロードがあります。
```
Comment (atoato88) [inline content/ja/docs/concepts/workloads/controllers/job.md:649]:
```suggestion
静的なインデックス付きJobの使用例としては、MPI、Horovord、Ray、PyTorchトレーニングジョブなど、インデックス付きJobのスケーリングを必要とするバッチワークロードがあります。
```

**Review by atoato88 — COMMENTED** [review]
Thank you for creating PR!
Almost LGTM, but I left some comments.
Please check those.

File: content/ja/docs/concepts/workloads/controllers/job.md [hunk 5]
```diff
@@ -251,10 +265,16 @@ Pod内のコンテナは、その中のプロセスが0以外の終了コード
 
 Podがノードからキックされた(ノードがアップグレード、再起動、削除されたなど)、または`.spec.template.spec.restartPolicy = "Never"`と設定されたときにPodに属するコンテナが失敗したなど、様々な理由でPod全体が故障することもあります。Podに障害が発生すると、Jobコントローラーは新しいPodを起動します。つまりアプリケーションは新しいPodで再起動された場合の処理を行う必要があります。特に、過去に実行した際に生じた一時ファイル、ロック、不完全な出力などを処理する必要があります。
 
+デフォルトでは、それぞれのPodの失敗は`.spec.backoffLimit`にカウントされます。しかし、[JobのPod失敗ポリシー](#pod-failure-policy)を設定することで、Pod失敗の処理をカスタマイズすることができます。
```
[the remark this answers is not in the document]
Comment (morix1500) [reply content/ja/docs/concepts/workloads/controllers/job.md]: Thank you @atoato88 !
I fixed it.

File: content/ja/docs/concepts/workloads/controllers/job.md (hunk 3 above)
Comment (morix1500) [reply content/ja/docs/concepts/workloads/controllers/job.md]: Thank you @atoato88 !
I fixed it.

File: content/ja/docs/concepts/workloads/controllers/job.md (hunk 4 above)
Comment (morix1500) [reply content/ja/docs/concepts/workloads/controllers/job.md]: Thank you @atoato88 !
I fixed it.

**Comment by atoato88** [comment]
/lgtm

Thank you for quick response!
ありがとうございます。

**Pull request merged**